### ISSUE TITLE
Anbox not working on ubuntu kernel

### ISSUE THREAD

**Issue by parithy** [anbox/anbox#1278]
Ashmem and binder modules built within the ubuntu kernel is not getting loaded when anbox is triggered
Even if it is manually loaded, anbox session-manager keeps complaining the modules are missing

**Comment by parithy** [comment]
Looks like binder module is not getting initialized. @zhsj Could you take a look at this?

**Comment by zhsj** [comment]
I don't have Ubuntu machine, so I'm not sure what's the root cause.

As I see from Debian, the built-in binder and ashmem work fine.

I think you should file bug to Ubuntu, you can follow steps on https://help.ubuntu.com/stable/ubuntu-help/report-ubuntu-bug.html.en

**Comment by parithy** [comment]
Could you confirm which kernel version it is when you say "As I see from Debian, the built-in binder and ashmem work fine."?

**Comment by zhsj** [comment]
> Could you confirm which kernel version it is when you say "As I see from Debian, the built-in binder and ashmem work fine."?

```
$ anbox system-info
version: 0.0~git20191022-1-Debian
cpu:
  arch:  x86
  brand: Intel(R) Core(TM) i7-6500U CPU @ 2.50GHz
  features: 
    - aes
    - sse4_1
    - sse4_2
    - avx
    - avx2
os:
  name: Debian GNU/Linux
  version: 
  snap-based: false
kernel:
  version: Linux version 5.3.0-1-amd64 (debian-kernel@lists.debian.org) (gcc version 9.2.1 20191008 (Debian 9.2.1-9)) #1 SMP Debian 5.3.7-1 (2019-10-19)
  binder: true
  ashmem: true
```

**Comment by parithy** [comment]
i see you have installed outside snap. I understand that  binder is a module and that snap container shouldn't be a problem. any chance that you are able to get this working in snap as well?

**Comment by zhsj** [comment]
> i see you have installed outside snap. I understand that binder is a module and that snap container shouldn't be a problem. any chance that you are able to get this working in snap as well?

Sorry I don't use snap. But I don't see there's any difference with or without snap.

**Comment by fengkaijia** [comment]
+1 binder doesn't load on Ubuntu 18.04 with 5.3 kernel.

```
version: 4
snap-revision: 158
cpu:
  arch:  x86
  brand: Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
  features: 
    - aes
    - sse4_1
    - sse4_2
    - avx
    - avx2
os:
  name: Ubuntu
  version: 18.04.3 LTS (Bionic Beaver)
  snap-based: true
kernel:
  version: Linux version 5.3.0-23-generic (buildd@lgw01-amd64-028) (gcc version 7.4.0 (Ubuntu 7.4.0-1ubuntu1~18.04.1)) #25~18.04.1-Ubuntu SMP Tue Nov 12 10:58:57 UTC 2019
  binder: false
  ashmem: true
```

**Comment by Mercurial** [comment]
same problem here

**Comment by fuzepod** [comment]
same here

**Comment by parithy** [comment]
This is fixed with pull request  #1309. Brief: anbox switch from out of tree modules of binder to Ubuntu kernel's built in binderfs
